### PR TITLE
add support for the `msync` syscall

### PR DESCRIPTION
**Pull request by maminrayej**

### CODE REVIEW

**Review by syrusakbary — COMMENTED** [review]
Looks good for me but either @john-sharratt or @theduke need to review and approve

File: libc-bottom-half/mman/mman.c
Comment (Arshia001) [inline libc-bottom-half/mman/mman.c:142]: Doesn't this function need to be declared in a header somewhere?

**Review by Arshia001 — CHANGES_REQUESTED** [review]
I'd wait for feedback from @john-sharratt before merging this in.

File: libc-bottom-half/mman/mman.c
Comment (maminrayej) [reply libc-bottom-half/mman/mman.c:142]: It is declared in `mman.h`.

Comment (Arshia001) [reply libc-bottom-half/mman/mman.c:142]: So what, the declaration was always there but the definition wasn't? Interesting!

**Review by theduke — APPROVED** [review]

File: libc-bottom-half/mman/mman.c
Comment (theduke) [inline libc-bottom-half/mman/mman.c:141]: Should probably be `> 0` to be on the safe side?
0 is not a valid FD.

**Review by theduke — APPROVED** [review]

**Pull request merged**
